Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -827,6 +831,57 @@ void AODProducerWorkflowDPL::fillMCTrackLabelsTable(const MCTrackLabelCursorType
   }
 }
 
+template <typename V0CursorType, typename CascadeCursorType>
+void AODProducerWorkflowDPL::fillSecondaryVertices(const o2::globaltracking::RecoContainer& recoData, V0CursorType& v0Cursor, CascadeCursorType& cascadeCursor)
+{
+
+  auto v0s = recoData.getV0s();
+  auto cascades = recoData.getCascades();
+
+  // filling v0s table
+  std::vector<bool> v0rej(v0s.size());
+  for (size_t i = 0; i < v0s.size(); i++) {
+    const auto& v0 = v0s[i];
+    auto trPosID = v0.getProngID(0);
+    auto trNegID = v0.getProngID(1);
+    int posTableIdx = -1;
+    int negTableIdx = -1;
+    auto item = mGIDToTableID.find(trPosID);
+    if (item != mGIDToTableID.end()) {
+      posTableIdx = item->second;
+    } else {
+      LOG(warn) << "Could not find a positive track index for prong ID " << trPosID;
+    }
+    item = mGIDToTableID.find(trNegID);
+    if (item != mGIDToTableID.end()) {
+      negTableIdx = item->second;
+    } else {
+      LOG(warn) << "Could not find a negative track index for prong ID " << trNegID;
+    }
+    if (posTableIdx != -1 and negTableIdx != -1) {
+      v0Cursor(0, v0.getVertexID(), posTableIdx, negTableIdx);
```

Comment:
`v0.getVertexID()` is the entry number in the collision table?

Reply:
you are right, for multi-TF processing this will create problems, will use mapping from vertexID to collisionID. But them I am not sure if the old code is correct: the reconstructed track indices are mapped to stored AOD index as https://github.com/AliceO2Group/AliceO2/blob/30327663640c088193254dd45d3e3a6f92040a1f/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L364-L365 (used to indicate V0's prong's indices). 
But they are reset after processing each TF:   https://github.com/AliceO2Group/AliceO2/blob/30327663640c088193254dd45d3e3a6f92040a1f/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L1475-L1476
It is OK to reset the map, since the reco indices are specific for each TF, but resetting the `mTableTrID` will break the correspondence to AOD track index (which is accumulated from one TF to another), right?

Reply:
Yes, it does not look sane.
Better in any case would be `mGIDToTableID.emplace(trackIndex, tracksCursor.lastIndex()); `

Reply:
@aalkin @ktf Is `lastIndex` not defined for the cursor. We use it in O2Physics task on something defined by `Produces`. How can one have the last insert id here?

Reply:
ok, I'll simply avoid resetting the counters like mTableTrID